src/neuroconv/datainterfaces/ecephys/basesortingextractorinterface.py

```diff
@@ -226,6 +226,38 @@ def subset_sorting(self):
         stub_sorting_extractor = self.sorting_extractor.frame_slice(start_frame=0, end_frame=end_frame)
         return stub_sorting_extractor
 
+    def add_recording_to_nwb(self, nwbfile: NWBFile, metadata: Optional[DeepDict] = None):
```

Comment:
```suggestion
    def add_ecephys_metadata_to_nwbfile(self, nwbfile: NWBFile, metadata: Optional[DeepDict] = None):
```

Original method name not entirely accurate; if we had a full recording attached, it would not be writing the traces data

Proposing to rename to explicitly reflect what it adds while also making it more similar to the new base `add_to_nwbfile` method name

Reply:
I agree that my choice is bad for the reasons that you describe. 
This is better.

I have a question though, isn't the units table also ecephys metadata and then this becomes overloaded?

Experimentally, what we are adding is the metadata of the recording device which is a combination of the recording device channels (that we add in the electrodes table), device and electrode groups. 

Can we do better or do you think is hard to disambiguate?

Reply:
OK, after thinking for a while I think that the best name is `add_channel_metadata_to_nwb` it covers exactly the type of metadata that we are adding. It is not a probe, is it nos electrodes, it is not channels as those concepts are kind of intermixed in the way that we are handilng them right now.

What we are doing now is to add the channel metadata that is a combination of the things above. It is all not ecephys because it is possible that there is other ecephys data that is not channel related.

Added this in the last commit.